crates/evm/execution-types/src/execution_outcome.rs

```diff
@@ -356,7 +380,7 @@ impl From<(BlockExecutionOutput<Receipt>, BlockNumber)> for ExecutionOutcome {
     fn from(value: (BlockExecutionOutput<Receipt>, BlockNumber)) -> Self {
         Self {
             bundle: value.0.state,
-            receipts: Receipts::from(value.0.receipts),
+            receipts: value.0.receipts.into_iter().map(Some).collect::<Vec<_>>().into(),
```

Comment:
this new syntax is more complicated, I'm missing the point of this change

Reply:
Before we had this implemented in reth

```rust
impl From<Vec<Receipt>> for Receipts {
    fn from(block_receipts: Vec<Receipt>) -> Self {
        Self { receipt_vec: vec![block_receipts.into_iter().map(Option::Some).collect()] }
    }
}
```

but now the situation is more generic in alloy with a generic parameter T https://github.com/alloy-rs/alloy/blob/86812dba8507e6f7741f666303a53cafef93c8c1/crates/consensus/src/receipt/receipts.rs#L81 so that this is not available anymore. But this should just be a temporary situation since we would like to remove soon the `Option<>` in the definition of `Receipts` in reth after the merge of this PR and some follow up work

Reply:
we have `Receipts<T>` in reth too now https://github.com/paradigmxyz/reth/issues/12358, https://github.com/paradigmxyz/reth/issues/12454

Reply:
The goal is was to use directly the alloy type, just importing it:
1. Use the Receipts alloy type everywhere with `pub type Receipts = alloy_consensus::Receipts<Option<Receipt>>;`
2. Do some work to arrive at `pub type Receipts = alloy_consensus::Receipts<Receipt>;` (especially on pruning which is blocker right now) 

But if we want to integrate a new Receipts type into reth maybe this PR is not relevant anymore

Reply:
I would hold off a bit with this pr until we have gotten further underway with https://github.com/paradigmxyz/reth/issues/12454